CEAligner module for structural alignment of 3d molecules

JoaoRodrigues opened the pull request:
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Biopython has 2 sequence-based structure alignment algorithms (SVDSuperimpose and QCPSuperimposer). This PR adds a structure-based structure alignment module, based on the CEalign code available in Pymol. The advantage of this algorithm is that it looks only at structure (in this specific case, distance matrices), and can find structural similarities where the sequence similarity is very very low.

The CEalign code in Pymol is written in C++, exposed to Python, and is licensed under a BSD-like license, which I believe is compatible with our BSD3 license. I kept the original headers and the original LICENSE file in the src folder.

Example API use:
```
from Bio.PDB import PDBParser
from Bio.PDB import CEAligner

parser = PDBParser()
s1 = parser.get_structure(...)  # one structure
s2 = parser.get_structure(...)  # another structure
s3 = parser.get_structure(...)  # a third structure

aligner = CEAligner()
aligner.set_reference(s1)
aligner.align(s2)  # this transforms s2
print(aligner.rms)

aligner.align(s3)  # this does not transform s3
print(aligner.rms)
```

mdehoon commented on Bio/PDB/cealign/ccealignmodule.cpp:
```diff
@@ -0,0 +1,747 @@
+
+  /* Unpack the arguments from Python */
+  if ( ! PyArg_ParseTuple(args, "OO|ii", &listA, &listB, &windowSize, &gap_max) ){
+      printf("Error parsing arguments for run_cealign\n");
```
This line can be removed; PyArg_ParseTUple will raise an exception if something goes wrong.

JoaoRodrigues commented:
@mdehoon, would you mind having a look at the C code? I removed most of the dependencies. Also replaced the RMSD calculations that were in the C code by calls to our own aligners, for simplicity (at a slight cost of performance).

JoaoRodrigues commented:
Oops, annoying rebase..

JoaoRodrigues commented:
Sorry for the multiple attempts. The test suite always passes on my laptop, so I have to debug through CI..

JoaoRodrigues commented:
@mdehoon @peterjc if there are no comments I'd like to go ahead and merge this :)

mdehoon commented:
@JoaoRodrigues  You may want to check if the C code is consistent with PEP 7, especially with regard to the C dialect.

peterjc commented:
I think you're right about the ``ccealignmodule.c`` license.

JoaoRodrigues commented:
> @JoaoRodrigues You may want to check if the C code is consistent with PEP 7, especially with regard to the C dialect.

TIL there is a PEP7. Although there is no specific PEP7 linter, I ran a couple of formatters/analyzers and made a few changes, namely regarding indentation, declaration formats, and public function naming. I compiled the extension with `-std=c89` and all I got were warnings about declarations in the for loops (e.g. `for (int i...)`), which from reading the PEP7 seem to be safe to ignore.

@peterjc thanks for the double-checking!

If the tests pass, I will proceed with the merging.
